hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/dml/TestMergeIntoTable2.scala

```diff
@@ -1030,4 +1031,97 @@ class TestMergeIntoTable2 extends HoodieSparkSqlTestBase {
       checkAnswer(s"show partitions $destTable")(Seq("dt=2022-09-26"))
     }
   }
+
+
+
+  test("Test MergeInto with various payloads") {
```

Comment:
Test both COW and MOR tables.  For MOR, set `hoodie.merge.small.file.group.candidates.limit=0`.